### PR TITLE
project setup

### PR DESCRIPTION
**Pull request by elijahladdie**
#7
## This PR is for project setup it contain initial project setup code for knights backend
- What is included in here ?
   - I created  utils folder that contain response utility we will be using for handling all responses 
   - All will be created in routes folder and get exported in the index file
   - All controller will be create in the controller folder and get exported in the index.ts file 
   - The same goes to services middleware utils and other functionalities that will be added in the program will be exported independently in the single file
 # How should be manually tested ? 
  - First clone this repository 
  ```bash 
  https://github.com/atlp-rwanda/knights-ecomm-be.git
  ```
  -  then to start development server use below command
  ```bash 
   npm run dev
  ```
  Happy coding!

### CODE REVIEW

**Comment by MC-Knight** [comment]
### Review

1. Follow best practices for NodeJs project structuring i.e config, controllers, , routes, middleware, ...

**Review by aimedivin — CHANGES_REQUESTED** [review]
Since we will be building a REST API and you're already using express.json() to parse JSON request bodies, you can remove the body-parser.

**Review by aimedivin — APPROVED** [review]

**Review by Ndevu12 — COMMENTED** [review]
Looks good but I can suggest to  place those @types under dependency instead of devDependency. It was the one that was ruining my deployment process. But I was using  "target": "NodeNext", in tsconfig.json.
You can leave it like that if we will not deploy it or if it work on "target": "es2016".

last Suggestion is full tsconfig set up. I can see that there is no "types"  and "typesRoots" property in tsconfig.

File: src/index.ts
Comment (GSinseswa721) [inline src/index.ts:1]: Great job ! You have followed best practice for setting up this project.

**Review by faid-terence — APPROVED** [review]
Nice Work @elijahladdie !

**Review by GSinseswa721 — APPROVED** [review]

**Comment by UwicyezaG** [comment]
The setup is great, also the structure is clean

**Pull request merged**